### PR TITLE
[libc][math] Fix overflow shifts for dyadic floats and add skip accuracy option for expm1.

### PR DESCRIPTION
**Pull request by lntue**
The following errors were reported by Paul Zimmerman:
```
expm1
Running worst cases check in --rndn mode...
FAIL x=0x1.c3b93d3b25c91p+8 ref=0x1.9fb21849affd1p+651 z=0x1.9f921849affd1p+651
Running worst cases check in --rndz mode...
FAIL x=0x1.d6479eba7c971p+8 ref=0x1.62a88613629b5p+678 z=0x1.62a886135e9b5p+678
Running worst cases check in --rndu mode...
FAIL x=0x1.1f0da93354198p+7 ref=0x1.0bd73b73fc74dp+207 z=0x1.0bd73b73fc74cp+207
Running worst cases check in --rndd mode...
FAIL x=0x1.93b3037166891p+8 ref=0x1.554f3a5c4535cp+582 z=0x1.554f3a5c4535bp+582
```
```
tan
Running worst cases check in --rndn mode...
FAIL x=0x1.00452f0e0134dp-13 ref=0x1.00452f2367da9p-13 z=0x1.00452f2367daap-13
Running worst cases check in --rndz mode...
FAIL x=0x1.89f0f5241255bp-2 ref=0x1.9e997dd861117p-2 z=0x1.9e997dd861118p-2
FAIL x=0x1.045457ae3994p+5 ref=0x1.0c06941768ef3p+1 z=0x1.0c06941768ef4p+1
```

### CODE REVIEW

**Comment by lntue** [comment]
@zimmermann6

File: libc/src/__support/FPUtil/dyadic_float.h [hunk 1]
```diff
@@ -260,10 +260,19 @@ LIBC_INLINE constexpr DyadicFloat<Bits> quick_add(DyadicFloat<Bits> a,
     return a;
 
   // Align exponents
-  if (a.exponent > b.exponent)
-    b.shift_right(a.exponent - b.exponent);
-  else if (b.exponent > a.exponent)
-    a.shift_right(b.exponent - a.exponent);
+  if (a.exponent > b.exponent) {
+    size_t shift = static_cast<size_t>(a.exponent - b.exponent);
+    if (shift < Bits)
```
Comment (michaelrj-google) [inline libc/src/__support/FPUtil/dyadic_float.h:265]: would it make sense to have this check in the `shift` functions instead of here?

**Comment by zimmermann6** [comment]
all expm1 tests from CORE-MATH are ok now, thanks!

**Comment by zimmermann6** [comment]
and all tan tests are ok too!

File: libc/src/__support/FPUtil/dyadic_float.h (hunk 1 above)
Comment (lntue) [reply libc/src/__support/FPUtil/dyadic_float.h]: Done.

**Review by jhuber6 — APPROVED** [review]

**Pull request merged**